darts/tests/models/forecasting/test_regression_models.py

```diff
@@ -1270,57 +1270,115 @@ def test_historical_forecast(self, mode):
-            ),
-            XGBModel(
-                lags=lags, output_chunk_length=2, multi_models=True, tree_method="exact"
-            ),
-            XGBModel(
-                lags=lags,
-                output_chunk_length=2,
-                multi_models=False,
-                tree_method="exact",
-            ),
-        ]
-        if lgbm_available:
-            models += [
-                LightGBMModel(lags=lags, output_chunk_length=1, multi_models=True),
-                LightGBMModel(lags=lags, output_chunk_length=1, multi_models=False),
-                LightGBMModel(lags=lags, output_chunk_length=2, multi_models=True),
-                LightGBMModel(lags=lags, output_chunk_length=2, multi_models=False),
-            ]
-        if cb_available:
-            models += [
-                CatBoostModel(lags=lags, output_chunk_length=1, multi_models=True),
-                CatBoostModel(lags=lags, output_chunk_length=1, multi_models=False),
-                CatBoostModel(lags=lags, output_chunk_length=2, multi_models=True),
-                CatBoostModel(lags=lags, output_chunk_length=2, multi_models=False),
-            ]
+    @pytest.mark.parametrize(
+        "config", itertools.product(model_configs, [1, 2], [True, False])
+    )
+    def test_multioutput_validation(self, config):
+        """Check that models not supporting multi-output are properly wrapped when ocl>1"""
+        (model_cls, model_kwargs), ocl, multi_models = config
         train, val = self.sine_univariate1.split_after(0.6)
+        model = model_cls(
+            **model_kwargs, lags=4, output_chunk_length=ocl, multi_models=multi_models
+        )
+        model.fit(series=train, val_series=val)
+        if model.output_chunk_length > 1 and model.multi_models:
+            assert isinstance(model.model, MultiOutputRegressor)
+        else:
+            assert not isinstance(model.model, MultiOutputRegressor)
 
-        for model in models:
-            model.fit(series=train, val_series=val)
-            if model.output_chunk_length > 1 and model.multi_models:
-                assert isinstance(model.model, MultiOutputRegressor)
+    def test_get_multioutput_estimator_multi_models(self):
+        """Craft training data so that estimator_[i].predict(X) == i + 1"""
+
+        def helper_check_overfitted_estimators(ts: TimeSeries, ocl: int):
+            m = XGBModel(lags=3, output_chunk_length=ocl, multi_models=True)
+            m.fit(ts)
+
+            assert len(m.model.estimators_) == ocl * ts.width
+
+            dummy_feats = np.array([[0, 0, 0] * ts.width])
+            for i in range(ocl):
+                for j in range(ts.width):
+                    sub_model = m.get_multioutput_estimator(horizon=i, target_dim=j)
+                    pred = sub_model.predict(dummy_feats)[0]
+                    # sub-model is overfitted on the training series
+                    assert np.abs(ts.width * i + j + 1 - pred) < 1e-2
```

Comment:
If we adapt the training examples a bit further below, we can simply check that the predictions are identical to the estimator counter
```suggestion
            estimator_counter = 0
            for i in range(ocl):
                for j in range(ts.width):
                    sub_model = m.get_multioutput_estimator(horizon=i, target_dim=j)
                    pred = sub_model.predict(dummy_feats)[0]
                    # sub-model is overfitted on the training series
                    assert np.abs(estimator_counter - pred) < 1e-2
                    estimator_counter += 1
```